### DISCUSSION TITLE
레벨3 아이디어 모음

### DISCUSSION THREAD

**Discussion by gracefulBrown** [woowacourse/prolog#906] [Ideas]
```
레벨3 목표
- 메타인지 향상을 위해 학습한 내용을 글로 정리할 수 있는 환경을 제공한다
- 나의 학습 역량을 스스로 증명할 수 있는 환경을 제공한다
```

- 레벨3의 목표를 기반으로 레벨3에 해결하고 싶은 문제나 개선하고 싶은 아이디어를 등록해주세요
- 다른사람이 등록된 아이디어를 공감하는 경우 이모지를 남기셔도 좋습니다.
- 레벨3 참여와 무관하게 아이디어를 등록하는 거라서 3기 멤버들도 자유롭게 등록 가능합니다.
- 정해진 형식은 없으니 자유롭게 남겨주세요~

**Comment by al-bur** [comment]
안녕하세요~ 프론트 4기 크루 앨버입니다. 프로필 스쿼드에서 한 번 얘기가 나온 적이 있었는데 괜찮은 것 같아 아이디어 공유합니다.

### 타겟 목표
- 나의 학습 역량을 스스로 증명할 수 있는 환경을 제공한다

### 추가하면 좋을 기능(Q&A 기능)
- 한 유저가 질문(개발 관련 내용등)을 등록하면, 해당 질문에 대해 다른 크루들이 답변을 해줄 수 있다.
- 답변 중 가장 많은 좋아요를 받은 댓글이 대표 답변으로 선정되어 보여진다.

### 기대효과
- 다른 크루의 질문에 답변을 하면서 유저들 스스로 학습 역량(자신이 얼마나 아는지)을 체크해볼 수 있다.
- 나의 답변이 대표 답변으로 선정되면 뿌듯하다😀

**Reply by wishoon** [reply]
호오 학습로그 스쿼드에서도 해당 아이디어가 나온 적 있었어요!!

Stack Overflow? 와 같은 느낌으로 질문을 등록하면, 답변을 등록하거나 학습로그를 작성해 태그하는 방식을 생각했었습니다 ㅎㅎ
또한, 현재 잘 활용되고 있지 않은 Slack의 #fe-4기-질문방, #be-4기-질문방을 알림방의 형태로 이용하는 것까지 아이디어가 나왔었습니다~~

**Comment by bigsaigon333** [comment]
가볍게(!) 아이디어 제안해보겠습니다..!

## 레벨3 목표

### 타겟 목표
- 프론트엔드 서비스의 안정성 증대

### 추가하면 좋을 기능
* JS -> TS 마이그레이션 100% 달성
* 런타임에서 json(api response) 스키마 검증
* Sentry 도입

### 기대 효과
- 타입안정성 증대로 인한 장애 가능성 저하 및 DX 향상
- Sentry 도입하여 client 에러에 신속히 대응
- 개개인의 TS 이해도 향상

**Reply by devhyun637** [reply]
TS 도입은 좋은데, Sentry까지는 무리가 있지 않을까 싶습니다.
그리고 전체 도입 보다는 스쿼드별로 가져가는 방법도 좋을 것 같아요! 서로 코드가 겹치면 conflict이 걱정됩니다

**Comment by devhyun637** [comment]
## 레벨3 목표
### 타겟 목표
역량 및 리포트 기능 리팩토링

### 보완해야하는 부분
역량 및 리포트 폴더구조 변경
역량 및 리포트 UI/UX 개선
역량 및 리포트 기능에 대한 툴팁을 두어 사용방법에 대한 가이드라인 제공
역량 및 리포트 타입스크립트 마이그레이션
(선택) e2e 테스트 도입

### 기대 효과
역량 및 리포트 사용 비율을 80%이상으로 늘리고, 이를 활용하여 모의면접에서 사용할 수 있도록 함
테스트 및 타입스크립트로 인한 안전성 추가

**Comment by bigsaigon333** [comment]
칭찬왕 등의 뱃지를 status 뱃지 형태로 아무 웹사이트에서나 embed 하게 하는 기능을 추가하는 것도 재밌을 것 같습니다~!

## ※ status뱃지 형태란?
![image](https://user-images.githubusercontent.com/31029000/176695706-f9d85b33-27a8-4740-ab2c-ef235fd99885.png)

**Comment by EastHShin** [comment]
가볍게 생각 한 번 해봤습니다ㅎㅎ
### 추가하면 좋을 기능
**명예의 전당 기능**

- 글 쓴 횟수에 따라 랭킹 등록
- 명예의 전당에 등록되지 않고 싶은 유저를 위해 노출 ON/OFF 기능 제공

### 기대효과
인기있는 학습로그 기능과 달리 랭킹 산정에 기간 범위 제한이 있지 않아 장기적인 동기부여 가능
